src/web-rtc-client.js

```diff
@@ -833,70 +834,106 @@ export default class WebRTCClient extends Emitter {
+        track.stop();
+      });
+    }
 
-      // $FlowFixMe
-      return navigator.mediaDevices.getUserMedia({ video: { deviceId: { exact: id } } }).then(async stream => {
-        const videoTrack = stream.getVideoTracks()[0];
-        const sender = pc && pc.getSenders().find(s => videoTrack && s && s.track && s.track.kind === videoTrack.kind);
+    const constraints = {
+      video: id ? { deviceId: { exact: id } } : true,
+    };
+    // $FlowFixMe
+    return navigator.mediaDevices.getUserMedia(constraints).then(async stream => {
+      const videoTrack = stream.getVideoTracks()[0];
+      let sender = pc && pc.getSenders().find(s => videoTrack && s && s.track && s.track.kind === videoTrack.kind);
+      if (!sender) {
+        sender = pc && pc.getSenders().find(s => !s.track);
+      }
 
-        if (sender) {
-          sender.replaceTrack(videoTrack);
-        }
+      if (sender) {
+        sender.replaceTrack(videoTrack);
+      }
 
-        // let's update the local stream
-        this._addLocalToVideoSession(this.getSipSessionId(session), stream);
-        this.eventEmitter.emit('onVideoInputChange', stream);
-        sdh.setLocalMediaStream(stream);
-        return stream;
-      });
-    }
+      // let's update the local stream
+      this._addLocalToVideoSession(this.getSipSessionId(session), stream);
+      this.eventEmitter.emit('onVideoInputChange', stream);
+      try {
+        await sdh.setLocalMediaStream(stream);
+      } catch (_) {
+        // Nothing to do
+      }
+      return stream;
+    });
   }
 
   getAudioDeviceId(): ?string {
     // $FlowFixMe
-    return this.audio && typeof this.audio === 'object' && 'deviceId' in this.audio ? this.audio.deviceId.exact : null;
+    return this.audio && typeof this.audio === 'object' && 'deviceId' in this.audio
+      ? this.audio.deviceId.exact : undefined;
   }
 
   getVideoDeviceId(): ?string {
     // $FlowFixMe
-    return this.video && typeof this.video === 'object' && 'deviceId' in this.video ? this.video.deviceId.exact : null;
+    return this.video && typeof this.video === 'object' && 'deviceId' in this.video
+      ? this.video.deviceId.exact : undefined;
   }
 
   changeVideo(enabled: boolean) {
     this.videoEnabled = enabled;
   }
 
-  reinvite(sipSession: Session, newConstraints: ?Object = null) {
+  reinvite(sipSession: Session, newConstraints: ?Object = null, conference: boolean = false) {
     if (newConstraints) {
       this.changeVideo(!!newConstraints.video);
     }
 
+    // When upgrading to video, remove the `deactivateVideoModifier` modifiers
+    if (newConstraints && newConstraints.video) {
+      const modifiers = sipSession.sessionDescriptionHandlerModifiersReInvite;
+      sipSession.sessionDescriptionHandlerModifiersReInvite = modifiers.filter(modifier =>
+        modifier !== deactivateVideoModifier);
+      sipSession.sessionDescriptionHandlerModifiersReInvite.push(activateVideoModifier);
```

Comment:
Are you sure we need an additional modifier? Isn't it enough to remove the `deactivateVideoModifier`?